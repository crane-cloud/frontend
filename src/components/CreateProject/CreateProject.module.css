.ProjectForm {
  width: 100%;
  font-size: 1rem;
}

.CreateProjectError {
  padding-bottom: 1rem;
  align-self: center;
}

.MainContentSection {
  padding-bottom: 1rem;
}

.FormElements {
  display: grid;
  grid-template-rows: 1fr 1fr 1fr;
}
.ContentSection {
  display: flex;
  flex-direction: column;
  overflow-x: auto;
  overflow-y: auto;
}
.InformationBar {
  background: #f8f7e5;
  height: 5rem;
  padding: 0 1rem;
  display: flex;
  align-items: center;
}

.InformationBarWithButton {
  flex: 1;
  display: flex;
  justify-content: space-between;
}

.InfoHeader {
  display: flex;
  align-items: center;
  font-weight: bold;
  font-size: 1.5rem;
}
.Element{
  display: flex;
  flex-direction: column;
  gap: 0.4rem;
}
.Element input {
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: 0.5fr 1fr;
  grid-gap: 0.2rem;
  width: 100%;
  border-radius: 5px;
}
.FormsSection {
  padding: 1rem;
  align-self: center;
  width: 60%;
}
.LeftFormElements {
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  gap: 1rem;
}

.TextArea {
  padding: 1rem;
  border: 1px solid black;
  box-sizing: border-box;
  font-size: inherit;
  width: 100%;
}
.TextArea::placeholder {
  color: #808080;
}

.WhiteBackground {
  color: #000;
}
.InitialBackground {
  color: #fff;
  background-color: transparent;
}
.CreateProjectError {
  display: flex;
  justify-content: center;
  align-items: center;
}

/* make it unresizable */
textarea {
  resize: none;
}

.ElementTitle {
  font-weight: bold;
  font-size: 0.9rem;

}
.InnerContent {
  display: flex;
  justify-content: center;
  gap: 1rem;
}

.InnerContent button {
  width: 100%;
}




.ClusterCheckboxSection {
  display: flex;
  padding: .4rem 0rem;
  flex-direction: row;
  font-size: 14px;
}

.NormalText{
  font-size: 1rem;
}

.ClusterToggleSection{
  display: flex;
  padding: .4rem .7rem;
  flex-direction: row;
  font-size: 14px;
  text-align: center;
  align-items: center;
}
.MultiSelectioOption {
  display: flex;
  padding: .4rem 4.9rem;
  font-size: 14px;
  color: #7c7a7a;
  text-align: center;
  align-items: center;
}
.Multipleclusters{
  display: flex;
  padding: .4rem 4.9rem;
}
li.ListStyle{
  list-style-type: none;
}
.leftsection{
  display: flex;
  align-items: center;
}
.ClusterLabel{
  display: flex;
  padding-left: 2rem;
}
@media(max-width: 650px) {

  .FormsSection {
    width: 90%;
  }

}