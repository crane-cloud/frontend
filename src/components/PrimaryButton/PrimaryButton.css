.Primary-Btn {
  background-color: #f7b21f;
  width: 9.7rem;
  height: 2.9rem;
  color: white;
  font-size: 1rem;
  font-weight: bold;
  line-height: 19px;
  text-align: center;
  border: none;
  outline: 1px solid #f7b21f;
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
}

.Primary-Btn:hover {
  background-color: transparent;
  color: #f7b21f;
  outline: 1px solid #f7b21f;
}

.Primary-Btn.CancelBtn {
  outline: 1px solid #000000;
  color: #000000;
  background-color: #ffffff;
}

.Primary-Btn.CancelBtn:hover {
  color: #fff;
  background-color: #000;
}
