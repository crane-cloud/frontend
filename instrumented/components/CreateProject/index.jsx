import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
// import { Redirect } from "react-router-dom";
import PrimaryButton from "../PrimaryButton";
import Select from "../Select";
//import CancelButton from "../CancelButton";
import Header from "../Header";
import Spinner from "../Spinner";
import Feedback from "../Feedback";
import BlackInputText from "../BlackInputText";
//import Checkbox from "../Checkbox";
import ToggleOnOffButton from "../ToggleOnOffButton";
//import { ReactComponent as InfoIcon } from "../../assets/images/info-icon.svg";
import styles from "./CreateProject.module.css";
import { handlePostRequestWithOutDataObject } from "../../apis/apis.js";
import { retrieveProjectTypes } from "../../helpers/projecttypes";
import handleProjectValidation from "../../helpers/validation";

class CreateProject extends React.Component {
  constructor(props) {
    super(props);
    const {
      clusters: { clusters },
    } = this.props;

    this.state = {
      projectName: "",
      clusterID: "",
      projectDescription: "",
      projectType: "",
      projectOrganisation: "",
      error: "",
      multiCluster: false,
      clusterchoices: false,
      othersBool: false,
      SelectedClusters: new Array(clusters?.length).fill(false),
      otherType: "",
      addingProject:false,
      addProjectError:"",
      addErrorCode:""
    };

    this.handleTypeSelectChange = this.handleTypeSelectChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.handleDatacenterSelectChange =
      this.handleDatacenterSelectChange.bind(this);
    this.changeMultiSelectioOption = this.changeMultiSelectioOption.bind(this);
    this.togglemultiCluster = this.togglemultiCluster.bind(this);
    this.handleOnChange = this.handleOnChange.bind(this);
    this.addNewProject  = this.addNewProject.bind(this);
  }

  // componentDidMount() {
    
  // }

  // componentDidUpdate(prevProps) {
   
  // }
  handleOnChange(position) {
    const { SelectedClusters } = this.state;
    this.setState({
      SelectedClusters: SelectedClusters.map((item, index) =>
        index === position ? !item : item
      ),
    });
  }
  changeMultiSelectioOption() {
    const { clusterchoices } = this.state;
    this.setState({
      clusterchoices: !clusterchoices,
    });
  }

  togglemultiCluster() {
    const { multiCluster } = this.state;
    this.setState({
      multiCluster: !multiCluster,
    });
  }

  handleTypeSelectChange(selected) {
    const { othersBool } = this.state;
    if (selected.id === 6) {
      if (!othersBool) {
        this.setState({ othersBool: true });
      }
    } else {
      this.setState({ projectType: selected.value });
      if (othersBool) {
        this.setState({ othersBool: false });
      }
    }
  }
  handleDatacenterSelectChange(selected) {
    this.setState({ clusterID: selected.id });
  }
  handleChange(e) {
    const { error, addProjectError } = this.state;
    this.setState({
      [e.target.name]: e.target.value,
    });

    if (error) {
      this.setState({
        error: "",
      });
    }
    if(addProjectError){
      this.setState({
        addProjectError: "",
      });
    }
  }

  handleSubmit() {
    const {
      projectName,
      projectDescription,
      clusterID,
      projectType,
      otherType,
      othersBool,
      projectOrganisation,
    } = this.state;
    const { data } = this.props;
    const capitalizeFirstLetter = (input) =>
      input.charAt(0).toUpperCase() + input.slice(1);
    const approvedType = othersBool
      ? capitalizeFirstLetter(otherType)
      : capitalizeFirstLetter(projectType);
    if (
      handleProjectValidation(
        projectName,
        projectDescription,
        approvedType,
        projectOrganisation,
        clusterID
      ) !== undefined
    ) {
      this.setState({
        error: handleProjectValidation(
          projectName,
          projectDescription,
          approvedType,
          projectOrganisation,
          clusterID
        ),
      });
    } else {
      const newProject = {
        description: projectDescription,
        cluster_id: clusterID,
        name: projectName,
        owner_id: data.id,
        organisation: projectOrganisation,
        project_type: approvedType,
      };
     // addProject(newProject);
     this.addNewProject(newProject)
    }
  }
  addNewProject(data) {
    this.setState({
      addingProject:true,
      addProjectError:""
    })
    handlePostRequestWithOutDataObject(
      data,
      `/projects`
    ).then(() => {
        window.location.href = `/projects`;
      })
      .catch((error) => {
        this.setState({
          addProjectError: "Failed to add project. Try again later",
          addingProject: false,
          addErrorCode:error.response?.status
        });
      });
  }

  render() {
    const {
      clusters: { clusters },
    } = this.props;
    const {
      projectName,
      projectDescription,
      error,
      projectOrganisation,
      othersBool,
      otherType,
      multiCluster,
      clusterchoices,
      SelectedClusters,
      addingProject,
      addProjectError,
      addErrorCode
    } = this.state;
    const types = retrieveProjectTypes();
   
    return (
      <div className={styles.MainContentSection}>
        <Header />
        <div className={styles.InformationBarSection}>
          <div className={styles.InformationBar}>
            <div className={styles.InformationBarWithButton}>
              <div className={styles.InfoHeader}>Create Project</div>
              <div className={styles.RoundAddButtonWrap}>
                {/*<CancelButton onClick={this.props.closeComponent} />*/}
                <PrimaryButton
                  btntype="close"
                  onClick={this.props.closeComponent}
                >
                  close
                </PrimaryButton>
              </div>
            </div>
          </div>
        </div>
        <div className={styles.ContentSection}>
          <div className={styles.FormsSection}>
            <div className="Card">
              <div className={styles.ProjectForm}>
                <div className={styles.LeftFormElements}>
                  <div className={styles.Element}>
                    <div className={styles.ElementTitle}>Datacenter</div>
                    <Select
                      required
                      placeholder="Choose Datacenter location"
                      options={clusters}
                      onChange={this.handleDatacenterSelectChange}
                    />
                    {/*  <div className={styles.ClusterCheckboxSection}>
                    <InfoIcon /> &nbsp; The above selection is for a single
                    cluster set up, for multi-cluster options check the box
                    below.
                  </div>
                  <div className={styles.ClusterCheckboxSection}>
                    <Checkbox
                      isBlack
                      onClick={this.togglemultiCluster}
                      isChecked={multiCluster}
                    />
                    <div className={styles.NormalText}>
                      &nbsp; Multi-cluster options
                    </div>
                  </div>*/}
                    {multiCluster && (
                      <div>
                        <div className={styles.ClusterToggleSection}>
                          <ToggleOnOffButton
                            onClick={this.changeMultiSelectioOption}
                          />{" "}
                          &nbsp; Cranecloud automatically selects the rest of
                          the clusters for this project.
                        </div>
                        {clusterchoices && (
                          <div>
                            <div className={styles.MultiSelectioOption}>
                              Please any other cluster for this project.
                            </div>
                            <div className={styles.Multipleclusters}>
                              {clusters.map(({ name, id }, index) => {
                                return (
                                  <li className={styles.ListStyle} key={index}>
                                    <div className={styles.clusterListItem}>
                                      <div className={styles.leftsection}>
                                        <input
                                          type="checkbox"
                                          id={id}
                                          name={name}
                                          value={name}
                                          checked={SelectedClusters[index]}
                                          onChange={() =>
                                            this.handleOnChange(index)
                                          }
                                        />
                                        <label
                                          className={styles.ClusterLabel}
                                          htmlFor={id}
                                        >
                                          {name}
                                        </label>
                                      </div>
                                    </div>
                                  </li>
                                );
                              })}
                            </div>
                          </div>
                        )}
                      </div>
                    )}
                  </div>

                  <div className={styles.Element}>
                    <div className={styles.ElementTitle}>Name</div>
                    <BlackInputText
                      required
                      placeholder="Project name"
                      name="projectName"
                      value={projectName}
                      onChange={(e) => {
                        this.handleChange(e);
                      }}
                    />
                  </div>
                  <div className={styles.Element}>
                    <div className={styles.ElementTitle}>Organisation</div>
                    <BlackInputText
                      required
                      placeholder="eg.Individual ,Makerere Universty..."
                      name="projectOrganisation"
                      value={projectOrganisation}
                      onChange={(e) => {
                        this.handleChange(e);
                      }}
                    />
                  </div>
                  <div className={styles.Element}>
                    <div className={styles.ElementTitle}>Type</div>
                    <Select
                      required
                      placeholder="Choose project type"
                      options={types}
                      onChange={this.handleTypeSelectChange}
                    />
                    {othersBool && (
                      <BlackInputText
                        required
                        placeholder="Type of project"
                        name="otherType"
                        value={otherType}
                        onChange={(e) => {
                          this.handleChange(e);
                        }}
                      />
                    )}
                  </div>
                  <div className={styles.Element}>
                    <div className={styles.ElementTitle}>Description</div>
                    <textarea
                      className={styles.TextArea}
                      type="text"
                      placeholder="Project description"
                      rows="4"
                      cols="50"
                      name="projectDescription"
                      value={projectDescription}
                      onChange={(e) => {
                        this.handleChange(e);
                      }}
                    />
                  </div>

                  {error && (
                    <div className={styles.CreateProjectError}>
                      <Feedback type="error" message={error} />
                    </div>
                  )}
                  {addProjectError && (
                    <div className={styles.CreateProjectError}>
                      <Feedback
                        message={
                          addErrorCode === 409
                            ? "Name already in use, please choose another"
                            : addProjectError
                        }
                        type={
                          addProjectError==="" && addErrorCode !== 409 ? "success" : "error"
                        }
                      />
                    </div>
                  )}
                  <div className={styles.InnerContent}>
                    <PrimaryButton
                      className="AuthBtn"
                      onClick={this.handleSubmit}
                      color="primary"
                    >
                      {addingProject ? <Spinner /> : "Create Project"}
                    </PrimaryButton>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

CreateProject.propTypes = {
  clusters: PropTypes.object,
  data: PropTypes.shape({
    id: PropTypes.string.isRequired,
  }).isRequired,
};

CreateProject.defaultProps = {
  clusters: [],
  params: {},
};

export const mapStateToProps = (state) => {
  const { data } = state.user;
  const { clusters } = state.clustersReducer;
  return {
    data,
    clusters,
  };
};

const mapDispatchToProps = {
};

export default connect(mapStateToProps, mapDispatchToProps)(CreateProject);
