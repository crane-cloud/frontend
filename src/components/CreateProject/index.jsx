import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import PrimaryButton from "../PrimaryButton";
import Select from "../Select";
import Header from "../Header";
import Spinner from "../Spinner";
import Feedback from "../Feedback";
import BlackInputText from "../BlackInputText";
import ToggleOnOffButton from "../ToggleOnOffButton";
import styles from "./CreateProject.module.css";
import { handlePostRequestWithOutDataObject } from "../../apis/apis.js";
import { retrieveProjectTypes } from "../../helpers/projecttypes";
import { namedOrganisations } from "../../helpers/projectOrganisations";
import handleProjectValidation from "../../helpers/validation";

class CreateProject extends React.Component {
  constructor(props) {
    super(props);
    const {
      clusters: { clusters },
    } = this.props;

    this.state = {
      projectName: "",
      clusterID: "",
      projectDescription: "",
      projectType: "",
      projectOrganisation: "",
      error: "",
      multiCluster: false,
      clusterchoices: false,
      othersBool: false,
      otherOrgType: false,
      SelectedClusters: new Array(clusters?.length).fill(false),
      otherType: "",
      addingProject: false,
      addProjectError: "",
      addErrorCode: "",
    };

    this.handleTypeSelectChange = this.handleTypeSelectChange.bind(this);
    this.handleOrganisationSelectChange =
      this.handleOrganisationSelectChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.handleDatacenterSelectChange =
      this.handleDatacenterSelectChange.bind(this);
    this.changeMultiSelectioOption = this.changeMultiSelectioOption.bind(this);
    this.togglemultiCluster = this.togglemultiCluster.bind(this);
    this.handleOnChange = this.handleOnChange.bind(this);
    this.addNewProject = this.addNewProject.bind(this);
  }

  handleOnChange(position) {
    const { SelectedClusters } = this.state;
    this.setState({
      SelectedClusters: SelectedClusters.map((item, index) =>
        index === position ? !item : item
      ),
    });
  }
  changeMultiSelectioOption() {
    const { clusterchoices } = this.state;
    this.setState({
      clusterchoices: !clusterchoices,
    });
  }

  togglemultiCluster() {
    const { multiCluster } = this.state;
    this.setState({
      multiCluster: !multiCluster,
    });
  }

  handleTypeSelectChange(selected) {
    const { othersBool } = this.state;
    if (selected.id === 6) {
      if (!othersBool) {
        this.setState({ othersBool: true });
      }
    } else {
      this.setState({ projectType: selected.value });
      if (othersBool) {
        this.setState({ othersBool: false });
      }
    }
  }

  handleOrganisationSelectChange(selected) {
    this.setState({ projectOrganisation: selected.value });
  }
  handleDatacenterSelectChange(selected) {
    this.setState({ clusterID: selected.id });
  }
  handleChange(e) {
    const { error, addProjectError } = this.state;
    this.setState({
      [e.target.name]: e.target.value,
    });

    if (error) {
      this.setState({
        error: "",
      });
    }
    if (addProjectError) {
      this.setState({
        addProjectError: "",
      });
    }
  }

  handleSubmit() {
    const {
      projectName,
      projectDescription,
      clusterID,
      projectType,
      otherType,
      othersBool,
      projectOrganisation,
    } = this.state;
    const { data } = this.props;
    const capitalizeFirstLetter = (input) =>
      input.charAt(0).toUpperCase() + input.slice(1);
    const approvedType = othersBool
      ? capitalizeFirstLetter(otherType)
      : capitalizeFirstLetter(projectType);
    if (
      handleProjectValidation(
        projectName,
        projectDescription,
        approvedType,
        projectOrganisation,
        clusterID
      ) !== undefined
    ) {
      this.setState({
        error: handleProjectValidation(
          projectName,
          projectDescription,
          approvedType,
          projectOrganisation,
          clusterID
        ),
      });
    } else {
      const newProject = {
        description: projectDescription,
        cluster_id: clusterID,
        name: projectName,
        owner_id: data.id,
        organisation: projectOrganisation,
        project_type: approvedType,
      };
      // addProject(newProject);
      this.addNewProject(newProject);
    }
  }
  addNewProject(data) {
    this.setState({
      addingProject: true,
      addProjectError: "",
    });
    handlePostRequestWithOutDataObject(data, `/projects`)
      .then(() => {
        window.location.href = `/projects`;
      })
      .catch((error) => {
        this.setState({
          addProjectError: "Failed to add project. Try again later",
          addingProject: false,
          addErrorCode: error.response?.status,
        });
      });
  }

  render() {
    const {
      clusters: { clusters },
    } = this.props;
    const {
      projectName,
      projectDescription,
      error,
      othersBool,
      otherType,
      multiCluster,
      clusterchoices,
      SelectedClusters,
      addingProject,
      addProjectError,
      addErrorCode,
    } = this.state;
    const types = retrieveProjectTypes();
    const presetOrganisations = namedOrganisations();

    return (
      <div className={styles.MainContentSection}>
        <Header />
        <div className={styles.InformationBarSection}>
          <div className={styles.InformationBar}>
            <div className={styles.InformationBarWithButton}>
              <div className={styles.InfoHeader}>Create Project</div>
              <div className={styles.RoundAddButtonWrap}>
                <PrimaryButton
                  btntype="close"
                  onClick={this.props.closeComponent}
                >
                  close
                </PrimaryButton>
              </div>
            </div>
          </div>
        </div>
        <div className={styles.ContentSection}>
          <div className={styles.FormsSection}>
            <div className="Card">
              <div className={styles.ProjectForm}>
                <div className={styles.LeftFormElements}>
                  <div className={styles.Element}>
                    <div className={styles.ElementTitle}>Datacenter</div>
                    <Select
                      required
                      placeholder="Choose Datacenter location"
                      options={clusters}
                      onChange={this.handleDatacenterSelectChange}
                    />
                    {multiCluster && (
                      <div>
                        <div className={styles.ClusterToggleSection}>
                          <ToggleOnOffButton
                            onClick={this.changeMultiSelectioOption}
                          />{" "}
                          &nbsp; Cranecloud automatically selects the rest of
                          the clusters for this project.
                        </div>
                        {clusterchoices && (
                          <div>
                            <div className={styles.MultiSelectioOption}>
                              Please any other cluster for this project.
                            </div>
                            <div className={styles.Multipleclusters}>
                              {clusters.map(({ name, id }, index) => {
                                return (
                                  <li className={styles.ListStyle} key={index}>
                                    <div className={styles.clusterListItem}>
                                      <div className={styles.leftsection}>
                                        <input
                                          type="checkbox"
                                          id={id}
                                          name={name}
                                          value={name}
                                          checked={SelectedClusters[index]}
                                          onChange={() =>
                                            this.handleOnChange(index)
                                          }
                                        />
                                        <label
                                          className={styles.ClusterLabel}
                                          htmlFor={id}
                                        >
                                          {name}
                                        </label>
                                      </div>
                                    </div>
                                  </li>
                                );
                              })}
                            </div>
                          </div>
                        )}
                      </div>
                    )}
                  </div>

                  <div className={styles.Element}>
                    <div className={styles.ElementTitle}>Name</div>
                    <BlackInputText
                      required
                      placeholder="Project name"
                      name="projectName"
                      value={projectName}
                      onChange={(e) => {
                        this.handleChange(e);
                      }}
                    />
                  </div>
                  <div className={styles.Element}>
                    <div className={styles.ElementTitle}>Organisation</div>
                    <Select
                      required
                      placeholder="Choose Organisation type"
                      options={presetOrganisations}
                      onChange={this.handleOrganisationSelectChange}
                    />
                  </div>
                  <div className={styles.Element}>
                    <div className={styles.ElementTitle}>Type</div>
                    <Select
                      required
                      placeholder="Choose project type"
                      options={types}
                      onChange={this.handleTypeSelectChange}
                    />
                    {othersBool && (
                      <BlackInputText
                        required
                        placeholder="Type of project"
                        name="otherType"
                        value={otherType}
                        onChange={(e) => {
                          this.handleChange(e);
                        }}
                      />
                    )}
                  </div>
                  <div className={styles.Element}>
                    <div className={styles.ElementTitle}>Description</div>
                    <textarea
                      className={styles.TextArea}
                      type="text"
                      placeholder="Project description"
                      rows="4"
                      cols="50"
                      name="projectDescription"
                      value={projectDescription}
                      onChange={(e) => {
                        this.handleChange(e);
                      }}
                    />
                  </div>

                  {error && (
                    <div className={styles.CreateProjectError}>
                      <Feedback type="error" message={error} />
                    </div>
                  )}
                  {addProjectError && (
                    <div className={styles.CreateProjectError}>
                      <Feedback
                        message={
                          addErrorCode === 409
                            ? "Name already in use, please choose another"
                            : addProjectError
                        }
                        type={
                          addProjectError === "" && addErrorCode !== 409
                            ? "success"
                            : "error"
                        }
                      />
                    </div>
                  )}
                  <div className={styles.InnerContent}>
                    <PrimaryButton
                      className="AuthBtn"
                      onClick={this.handleSubmit}
                      color="primary"
                    >
                      {addingProject ? <Spinner /> : "Create Project"}
                    </PrimaryButton>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

CreateProject.propTypes = {
  clusters: PropTypes.object,
  data: PropTypes.shape({
    id: PropTypes.string.isRequired,
  }).isRequired,
};

CreateProject.defaultProps = {
  clusters: [],
  params: {},
};

export const mapStateToProps = (state) => {
  const { data } = state.user;
  const { clusters } = state.clustersReducer;
  return {
    data,
    clusters,
  };
};

const mapDispatchToProps = {};

export default connect(mapStateToProps, mapDispatchToProps)(CreateProject);
