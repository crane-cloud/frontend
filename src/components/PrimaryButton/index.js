import React from "react";
import "./PrimaryButton.css";
import PropTypes from "prop-types";

const PrimaryButton = (props) => {
  const { children, className, btntype, color, small, transparent, noPadding } = props;

  const getColorClass = () => {
    switch (color) {
      case "primary":
        return "PrimaryColorBtn";
      case "primary-outline":
        return "PrimaryOutlineColorBtn";
      case "black":
        return "BlackPrimaryBtn";
      case "red":
        return "PrimaryDeleteBtn";
      case "red-outline":
        return "PrimaryDeleteBtnOutline";
      default:
        return "primary-button--blue";
    }
  };
  return (
    <button
      {...props}
      className={`Primary-Btn ${className} ${btntype === "close" && "DeleteBtnOutline"}
        ${color && getColorClass()} ${small===true ? "SmallBtn" : ""} ${transparent && "TransparentBtn"}
        ${noPadding && "NoPaddingBtn"}
      `}
    >
      {children}
    </button>
  );
};
PrimaryButton.propTypes = {
  small: PropTypes.bool,
};
PrimaryButton.defaultProps = {
  small: false,
};

export default PrimaryButton;
