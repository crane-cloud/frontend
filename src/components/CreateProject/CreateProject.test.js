import React from "react";
import { shallow } from "enzyme";
import { useSelector } from "react-redux";
import CreateProject from "./";

import PrimaryButton from "../PrimaryButton";

jest.mock("react-redux", () => ({
  useSelector: jest.fn(),
}));

describe("CreateProject Component", () => {
  let closeComponent;

  beforeEach(() => {
    closeComponent = jest.fn();
    // Mock the redux state
    useSelector.mockImplementation((selector) =>
      selector({
        clustersReducer: {
          clusters: [
            { id: "1", name: "Cluster1" },
            { id: "2", name: "Cluster2" },
          ],
        },
        user: {
          data: { id: "user1" },
        },
      })
    );
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("should render the component correctly", () => {
    const wrapper = shallow(<CreateProject closeComponent={closeComponent} />);

    expect(wrapper.find(PrimaryButton).exists()).toBe(true);
  });
  it("matches the Header snapshot", () => {
    const wrapper = shallow(<CreateProject closeComponent={closeComponent} />);
    expect(wrapper).toMatchSnapshot();
  });

});
