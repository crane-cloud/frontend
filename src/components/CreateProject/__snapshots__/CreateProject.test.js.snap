// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateProject Component matches the Header snapshot 1`] = `ShallowWrapper {}`;

exports[`Testing the App Metrics Page component matchs the CreateProject component snapshot 1`] = `ShallowWrapper {}`;
